contents/product-engineers/dogfooding.md

```diff
@@ -0,0 +1,137 @@
+  - Engineering
+  - Feature management
+crosspost:
+  - Blog
+---
+
+Dogfood is not on PostHog's roadmap, but it sure does shape it. 
+
+Dogfooding is the process of using your product internally. This helps you to get fast feedback, ship a higher-quality product, and maintain customer empathy.
+
+At PostHog, we dogfood a lot and find it works well because we're our [own ideal customer](/newsletter/ideal-customer-profile-framework). We  build tools for companies that look like ours. Not all companies are in this situation though, meaning dogfooding can lead you in the wrong direction if you aren't careful.
+
+To help you figure out if dogfooding is right for your company and how to do it well, this post covers our experiences, how we implement dogfooding, and pitfalls to avoid.
+
+## What does dogfooding look like at PostHog?
+
+As a high-growth startup making a tool for high-growth startups, we end up using our product a lot. Dogfooding doesn't just happen with the product we've already built, but also what is in progress too. In this way, it is part of our entire product development process.
+
+To give you an idea of how dogfooding shapes PostHog, here are some examples:
+
+### How booking user interviews led to surveys
+
+Our product team found booking user interviews laborious. They needed to identify users, send them emails, and coordinate a time before getting any feedback. Across hundreds of interviews, this takes up a significant amount of time.
+
+To solve this, they decided to leverage our site app functionality to [build a popup](/tutorials/feedback-interviews-site-apps) that used feature flags to target the right users with a button to book a meeting in Calendly.
+
+![Surveys](https://res.cloudinary.com/dmukukwp6/image/upload/survey_c020075d73.png)
+
+This saved them a massive amount of time while also validating an important use case for what is now [surveys](/surveys). In this way, dogfooding helped us both at that moment and guided our roadmap in the longer term.
+
+### HogQL maximalism
+
+For a time, you couldn't query your PostHog data directly with SQL. That changed with [HogQL](/docs/hogql), our version of SQL (and an entire querying service). 
+
+The development of HogQL was largely driven by dogfooding for two reasons:
+
+1. **It's advanced.** Because HogQL is a tool for power users, it was tougher to get initial adoption. We were the primary ones using it meaning we were the ones finding issues and limitations. 
+
+2. **We use it elsewhere.** HogQL is now the underlying query engine for all insights. Improvements to functionality and performance we created internally end up benefiting external users too.
+
+For example, HogQL was missing `OFFSET` support until Michael discovered that limitation when wanting to calculate the average number of times a dashboard is viewed. 
+
+![Michael dogfooding](https://res.cloudinary.com/dmukukwp6/image/upload/dogfood_2b527fee65.png)
+
+### Everyone needs to query external sources
```

Comment:
this example feels a little long. maybe trim down the intro line(s) and look to cut back a bit elsewhere